### DISCUSSION TITLE
Is there a way to have global warnings/constraints show up in the rendered output?

### DISCUSSION THREAD

**Discussion by HughG** [useblocks/sphinx-needs#924] [Q&A]
Firstly, thanks for this great piece of software, and its related ecosystem :-)  I've started experimenting with the VSCode extension, and would like to play with the modelling library as well, when I get time ...

I'm wondering whether there's an easy (or not-so-easy) way to have something like the `needs_warnings` or `needs_constraints` but have rules which apply to all needs, and appear in the rendered output (specifically, HTML).  It seems that warnings apply to all needs, but only appear in the build output, whereas constraints appear in the rendered output but only for needs which explicitly mention those constraints.

Bear in mind that I'm an experienced developer but a Python novice.

**Comment by danwos** [comment]
You can use [global options](https://sphinx-needs.readthedocs.io/en/latest/configuration.html#needs-global-options) to define values for needs globally. So all needs or a subset of it can get the needed constraints, without writing this down for each need by hand.

**Reply by HughG** [reply]
Hi @danwos.  As I commented elsewhere, I'm not like to spend more time on this any time soon.  But I still think it would be useful to have a default set of constraints to apply to all needs.
